content/00.build/65.developer-reference/40.account-abstraction/40.building-smart-accounts.md

```diff
@@ -28,14 +28,27 @@ For this purpose, use the `createAccount` or `create2Account` methods of the dep
 system contract, rather than the general `create` or `create2` methods, which are
 used for contracts not intended to act as accounts.
 
-#### Example Using `zksync-ethers` SDK (v5)
+#### Example Using `zksync-ethers` SDK (v6)
 
 ```ts
-import { ContractFactory } from "zksync-ethers";
-
-const contractFactory = new ContractFactory(abi, bytecode, initiator, "createAccount");
-const aa = await contractFactory.deploy(...args);
-await aa.deployed();
+async function deployAAFactory(hre: HardhatRuntimeEnvironment) {
+  const deployer = hre.deployer;
+  // the factory contract
+  const factoryArtifact = await deployer.loadArtifact('AAFactory');
+  // the account contract
+  const aaArtifact = await deployer.loadArtifact('Account');
+
+  const factory = await deployer.deploy(
+    factoryArtifact,
+    [utils.hashBytecode(aaArtifact.bytecode)],
+    undefined,
+    undefined,
+    [aaArtifact.bytecode]
+  );
+  const factoryAddress = await factory.getAddress();
+  console.log(`AA factory address: ${factoryAddress}`);
+  return factory;
+}
```

Comment:
Any particular reason we dont showcase / use `SmartAccount` [class](https://sdk.zksync.io/js/ethers/api/v6/accounts/smartaccount) here?

Reply:
it doesn't seem like there is a way to use that for deploying afaik, but I added another example about the [account factories](https://sdk.zksync.io/js/ethers/api/v6/accounts/smartaccount-factories)